File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        return speed switch
        {
            0 => 0.0,
            <= 4 => 1.0,
            <= 8 => 0.9,
            9 => 0.8,
            _ => 0.77
        };
    }

    public static double ProductionRatePerHour(int speed) 
        => speed * 221.0 * (SuccessRate(speed));

    public static int WorkingItemsPerMinute(int speed) =>
        (int)(ProductionRatePerHour(speed) / 60.0);
}

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        int carsPerHour = 221 * speed;
        if (speed <= 4) {
            return carsPerHour;
        }
        if (speed <= 8) {
            return carsPerHour * 0.9;
        }
        return (speed == 9) ? carsPerHour * 0.8 : carsPerHour * 0.77;
    }

    public int workingItemsPerMinute(int speed) {
        final double perMin = productionRatePerHour(speed) / 60;
        return (int) Math.floor(perMin);
    }
}
